### ISSUE TITLE
Parse error for at-content without a trailing semicolon

### ISSUE THREAD

**Issue by jamesarosen** [stylelint/stylelint#2185]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

The following construct, which is valid SCSS, is labeled `Unclosed block CssSyntaxError`:

```scss
@mixin myMixin {
  .some-selector {@content}
}
```

The following version, which is semantically identical, is OK in stylelint:

```scss
@mixin myMixin {
  .some-selector {
    @content;
  }
}
```

> Which rule, if any, is this issue related to?

n/a

> What CSS is needed to reproduce this issue?

N/A

> What stylelint configuration is needed to reproduce this issue?

My `.stylelintrc` is

```json
{
  "rules": {
    "block-no-empty": true,
    "color-hex-case": "upper",
    "color-hex-length": "long",
    "color-named": "never",
    "color-no-invalid-hex": true,
    "comment-empty-line-before": [ "always", {
      "ignore": ["stylelint-commands", "between-comments"]
    } ],
    "declaration-bang-space-after": "never",
    "declaration-colon-space-after": "always",
    "declaration-bang-space-before": "always",
    "font-family-name-quotes": "always-where-recommended",
    "indentation": [ 2, {
      "ignore": [ "inside-parens", "value" ]
    }],
    "max-empty-lines": 2,
    "max-nesting-depth": 5,
    "number-leading-zero": "always",
    "number-max-precision": 3,
    "number-no-trailing-zeros": true,
    "property-case": "lower",
    "property-no-vendor-prefix": true,
    "rule-nested-empty-line-before": [ "always-multi-line", {
      "except": ["first-nested"],
      "ignore": ["after-comment"]
    }],
    "string-no-newline": true,
    "string-quotes": "single",
    "unit-case": "lower",
    "unit-no-unknown": true,
    "unit-whitelist": ["%", "deg", "em", "ms", "pt", "px", "rem", "s", "vh", "vw"],
    "value-keyword-case": "lower",
    "value-no-vendor-prefix": true
  }
}
```

> Which version of stylelint are you using?

7.5.0, from https://github.com/billybonks/broccoli-stylelint/tree/82a79754501c8502aeba3d35d2d3d0d1e3339ff1

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`ember build`, using https://github.com/billybonks/ember-cli-stylelint, which uses https://github.com/billybonks/broccoli-stylelint

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, related to SCSS mixins.

> What did you expect to happen?

No error for that block *or* some sort of violation related to spacing or lines around blocks.

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
app/styles/my.scss
 2:32  ✖  Unclosed block   CssSyntaxError
```

**Comment by Arcanemagus** [comment]
`postcss-scss` was updated to v0.4.0 in v7.6.0, do you still see this with that version?

**Comment by jamesarosen** [comment]
v7.6.0 doesn't seem to help:
```shell
$ npm install stylelint@7.6.0
$ ember build
...
app/styles/fastly/mixins/_placeholder.scss
 1:1  ✖  Unclosed block   CssSyntaxError
```
Do you think postcss-scss is the right place to raise the issue?

**Comment by jeddy3** [comment]
> Do you think postcss-scss is the right place to raise the issue?

Yes. Please raise an issue there. It'll probably help them out if you can find some reference to `@content` _without a trailing semicolon_ being valid according to the SCSS spec, as the following is parsed correctly by `postcss-scss`:

```scss
@mixin myMixin {
  .some-selector {@content;}
}
```

**Comment by jamesarosen** [comment]
> the following is parsed correctly by postcss-scss

Great catch! libsass handles the semicolon-less version without any warnings, but that doesn't mean it's valid per the spec.
